### PR TITLE
Feat/remove contribution

### PR DESCRIPTION
**Pull request by oxlime**
This is what i have so far, my idea of removing by basically zero'ing out the contribution. I'm having issues with removing while also looping through the contributions.

### CODE REVIEW

File: contracts/onlydust/deathnote/core/contributions/test_contributions.cairo [hunk 1]
```diff
@@ -53,6 +53,33 @@ func test_new_contribution_can_be_added{
...
+    let (count, contribs) = contributions.all_contributions()
+
+    assert 2 = count
+
+    %{ stop_prank = start_prank(ids.FEEDER) %}
+    let (local contribution1) = contributions.remove_contribution(123)
+    %{ stop_prank() %}
+
+    let (count, contribs) = contributions.all_contributions()
+
```
Comment (AnthonyBuisset) [inline contracts/onlydust/deathnote/core/contributions/test_contributions.cairo:78]: you can also check that if you try to read the deleted contribution, the call is reverted with "Contributions: Contribution does not exist"

**Review by AnthonyBuisset — CHANGES_REQUESTED** [review]
Nice ! Thanks for the contribution.

Could you please modify the e2e test as well ? Check `test_contributions.e2e.cairo` file.
It should show that a change is missing in the Interface as well.

I do not think there is a better way of doing it as it is not possible to delete storage in starknet AFAIK. Zero-ing all fields is the way to do.
Good work overall !

**Comment by oxlime** [comment]
Deleted contributions are still indexed but all values set to 0. So I reworked the fetch_contribution_loop to loop over contributions with contribution_id = 0.

**Comment by oxlime** [comment]
closing this because i opened a new pr after the refactor

**Pull request closed without merging**